include/micm/configure/solver_config.hpp

```diff
@@ -361,51 +366,66 @@ namespace micm
       const std::string THIRD_BODY = "THIRD_BODY";
 
       ValidateSchema(object, { NAME, TYPE }, { TRACER_TYPE, ABS_TOLERANCE, DIFFUSION_COEFF, MOL_WEIGHT });
-
-      std::string name = object[NAME].get<std::string>();
+      
+      std::string name = object[NAME].as<std::string>();
       Species species{ name };
 
       // Load remaining keys as properties
-      for (auto& [key, value] : object.items())
+      for(auto it = object.begin(); it != object.end(); ++it)
       {
+        auto key = it->first.as<std::string>();
+        auto value = it->second;
+
         if (key != NAME && key != TYPE)
-        {
-          if (value.is_string())
+        {       
+          std::string stringValue = value.as<std::string>();
+          if (!stringValue.empty() && isInt(stringValue)) {
+            species.SetProperty<int>(key, value.as<int>());
+          }
+          else if (!stringValue.empty() && isFloat(stringValue)) {
+            species.SetProperty<double>(key, value.as<double>());
+          }
+          else if (!stringValue.empty() && isBool(stringValue))
+          {
+            species.SetProperty<bool>(key, value.as<bool>());
+          }
+          else
           {
-            if (key == TRACER_TYPE && value == THIRD_BODY)
+            if (key == TRACER_TYPE && stringValue == THIRD_BODY)
             {
               species.SetThirdBody();
             }
             else
             {
-              species.SetProperty<std::string>(key, value);
+              species.SetProperty<std::string>(key, stringValue);
             }
           }
-          else if (value.is_number_integer())
-          {
-            species.SetProperty<int>(key, value);
-          }
-          else if (value.is_number_float())
-          {
-            species.SetProperty<double>(key, value);
-          }
-          else if (value.is_boolean())
-          {
-            species.SetProperty<bool>(key, value);
-          }
-          else
-          {
-            throw std::system_error{ make_error_code(MicmConfigErrc::InvalidType), key };
-          }
-        }
+        } 
       }
       species_[name] = species;
+    }        
+
+    // Utility functions to check types and perform conversions
+    bool isBool(const std::string& value) {
+        return (value == "true" || value == "false");
```

Comment:
I am not sure where to put these functions.

Reply:
I think putting them below all the parsing functions would be good for consistency. We use CamelCase for function names (refers to our [clang tidy](https://github.com/NCAR/micm/blob/main/.clang-tidy)). Could you modify them in CamelCase?
eg) `isBool` ->`IsBool()`